File: app/src/main/java/com/antonchaynikov/triptracker/data/location/LocationSourceModule.java
package com.antonchaynikov.triptracker.data.location;

import android.content.Context;

import dagger.Module;
import dagger.Provides;

@Module(includes = LocationProviderModule.class)
public class LocationSourceModule {

    @Provides
    public LocationSource provideLocationSource(Context context, LocationProvider locationProvider) {
        LocationSourceImpl locationSource = LocationSourceImpl.getInstance(context);
        locationSource.setLocationProvider(locationProvider);
        return LocationSourceImpl.getInstance(context);
    }
}

File: core/src/main/java/com/antonchaynikov/core/data/location/LocationSourceModule.kt
package com.antonchaynikov.core.data.location

import android.content.Context
import com.antonchaynikov.core.data.location.LocationSource
import com.antonchaynikov.core.data.location.LocationSourceImpl
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module(includes = [LocationProviderModule::class])
class LocationSourceModule {
    @Singleton
    @Provides
    fun locationSource(context: Context, locationProvider: LocationProvider): LocationSource {
        val locationSource = LocationSourceImpl.getInstance(context)
        locationSource.setLocationProvider(locationProvider)
        return LocationSourceImpl.getInstance(context)
    }
}

